Sharding-proxy 4.0.0 can query columns for table

## Bug Report

When query columns of sharding table by `getColumns` method of JDBC, sharding-proxy will throw exception:
```
Exception in thread "main" java.sql.SQLException: 2Unknown exception: [Cannot find table rule and default data source with logic table: '%']
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1086)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:4237)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:4169)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2617)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2778)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2828)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2777)
	at com.mysql.jdbc.StatementImpl.executeQuery(StatementImpl.java:1651)
	at com.mysql.jdbc.DatabaseMetaData$2.forEach(DatabaseMetaData.java:2711)
	at com.mysql.jdbc.DatabaseMetaData$2.forEach(DatabaseMetaData.java:2598)
	at com.mysql.jdbc.IterateBlock.doForAll(IterateBlock.java:50)
	at com.mysql.jdbc.DatabaseMetaData.getColumns(DatabaseMetaData.java:2597)
	at jdbc.JDBCUtil.main(JDBCUtil.java:38)
```

But I used sharding-proxy 4.0.0-RC2, It can work well.

### Which version of ShardingSphere did you use?

4.0.0

### Which project did you use? Sharding-JDBC or Sharding-Proxy?

Sharding-Proxy

### Expected behavior

get Column info from Result Set.

### Actual behavior

Throw exception.

### Reason analyze (If you can)

the method `getColumn` should map to SQL `SHOW FULL COLUMNS ...`

So I think there are some error in `parse` or `extractor`.

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

```yaml
schemaName: sharding_db

dataSources:
 ds_0:
   url: jdbc:mysql://127.0.0.1:3306/demo_ds_0?serverTimezone=UTC&useSSL=false
   username: root
   password:
   connectionTimeoutMilliseconds: 30000
   idleTimeoutMilliseconds: 60000
   maxLifetimeMilliseconds: 1800000
   maxPoolSize: 50
 ds_1:
   url: jdbc:mysql://127.0.0.1:3306/demo_ds_1?serverTimezone=UTC&useSSL=false
   username: root
   password:
   connectionTimeoutMilliseconds: 30000
   idleTimeoutMilliseconds: 60000
   maxLifetimeMilliseconds: 1800000
   maxPoolSize: 50

shardingRule:
 tables:
   t_order:
     actualDataNodes: ds_${0..1}.t_order_${0..1}
     tableStrategy:
       inline:
         shardingColumn: order_id
         algorithmExpression: t_order_${order_id % 2}
     keyGenerator:
       type: SNOWFLAKE
       column: order_id
   t_order_item:
     actualDataNodes: ds_${0..1}.t_order_item_${0..1}
     tableStrategy:
       inline:
         shardingColumn: order_id
         algorithmExpression: t_order_item_${order_id % 2}
     keyGenerator:
       type: SNOWFLAKE
       column: order_item_id
 bindingTables:
   - t_order,t_order_item
 defaultDatabaseStrategy:
   inline:
     shardingColumn: user_id
     algorithmExpression: ds_${user_id % 2}
 defaultTableStrategy:
   none:

```

### Example codes for reproduce this issue (such as a github link).

```java
        Class.forName("com.mysql.jdbc.Driver");
        try (Connection connection = DriverManager.getConnection("jdbc:mysql://localhost:3307/sharding_db", "root", "root");
            Statement statement = connection.createStatement()) {
            ResultSet rs = metaData.getColumns("sharding_db", null, "t_order", "%");
       }
```

```
[INFO ] 15:27:04.980 [ShardingSphere-Command-4] ShardingSphere-SQL - Rule Type: sharding
[INFO ] 15:27:04.981 [ShardingSphere-Command-4] ShardingSphere-SQL - Logic SQL: SHOW FULL TABLES FROM `sharding_db` LIKE 't_order'
[INFO ] 15:27:04.981 [ShardingSphere-Command-4] ShardingSphere-SQL - SQLStatement: CommonSQLStatementContext(sqlStatement=org.apache.shardingsphere.sql.parser.sql.statement.dal.dialect.mysql.ShowTablesStatement@7d8a9fd6, tablesContext=TablesContext(tables=[Table(name=t_order, alias=Optional.absent())], schema=Optional.absent()))
[INFO ] 15:27:04.981 [ShardingSphere-Command-4] ShardingSphere-SQL - Actual SQL: ds_0 ::: SHOW FULL TABLES  LIKE 't_order_0'
[ERROR] 15:27:04.993 [ShardingSphere-Command-5] o.a.s.s.f.c.CommandExecutorTask - Exception occur: 
org.apache.shardingsphere.core.config.ShardingConfigurationException: Cannot find table rule and default data source with logic table: '%'
	at org.apache.shardingsphere.core.rule.ShardingRule.getTableRule(ShardingRule.java:191)
	at org.apache.shardingsphere.core.route.type.unicast.UnicastRoutingEngine.route(UnicastRoutingEngine.java:78)
	at org.apache.shardingsphere.core.route.router.sharding.ShardingRouter.route(ShardingRouter.java:113)
	at org.apache.shardingsphere.core.route.StatementRoutingEngine.route(StatementRoutingEngine.java:54)
	at org.apache.shardingsphere.core.SimpleQueryShardingEngine.route(SimpleQueryShardingEngine.java:58)
	at org.apache.shardingsphere.core.BaseShardingEngine.executeRoute(BaseShardingEngine.java:85)
	at org.apache.shardingsphere.core.BaseShardingEngine.shard(BaseShardingEngine.java:68)
	at org.apache.shardingsphere.shardingproxy.backend.communication.jdbc.wrapper.StatementExecutorWrapper.doShardingRoute(StatementExecutorWrapper.java:79)
	at org.apache.shardingsphere.shardingproxy.backend.communication.jdbc.wrapper.StatementExecutorWrapper.route(StatementExecutorWrapper.java:65)
	at org.apache.shardingsphere.shardingproxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:83)
	at org.apache.shardingsphere.shardingproxy.backend.text.admin.UnicastBackendHandler.execute(UnicastBackendHandler.java:57)
	at org.apache.shardingsphere.shardingproxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:73)
	at org.apache.shardingsphere.shardingproxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:92)
	at org.apache.shardingsphere.shardingproxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:72)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4
+++ b/shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4
@@ -40,7 +40,7 @@ showTableStatus
     ;
 
 showColumns
-    : SHOW EXTENDED? FULL? (COLUMNS | FIELDS) fromTable fromSchema? identifier*
+    : SHOW EXTENDED? FULL? (COLUMNS | FIELDS) fromTable fromSchema?  (showLike_ | showWhereClause_)?
     ;
 
 showIndex
@@ -64,6 +64,10 @@ fromTable
     ;
 
 showLike
+    : LIKE stringLiterals
+    ;
+
+showLike_
     : LIKE stringLiterals
     ;
 

--- a/shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4
+++ b/shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4
@@ -40,7 +40,7 @@ showTableStatus
     ;
 
 showColumns
-    : SHOW EXTENDED? FULL? (COLUMNS | FIELDS) fromTable fromSchema?  (showLike_ | showWhereClause_)?
+    : SHOW EXTENDED? FULL? (COLUMNS | FIELDS) fromTable fromSchema?  (showColumnLike_ | showWhereClause_)?
     ;
 
 showIndex
@@ -67,7 +67,7 @@ showLike
     : LIKE stringLiterals
     ;
 
-showLike_
+showColumnLike_
     : LIKE stringLiterals
     ;
 

--- a/shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4
+++ b/shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/antlr4/imports/mysql/DALStatement.g4
@@ -40,7 +40,7 @@ showTableStatus
     ;
 
 showColumns
-    : SHOW EXTENDED? FULL? (COLUMNS | FIELDS) fromTable fromSchema? (showLike | showWhereClause_)?
+    : SHOW EXTENDED? FULL? (COLUMNS | FIELDS) fromTable fromSchema? identifier*
     ;
 
 showIndex

--- a/shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/visitor/MySQLDALVisitor.java
+++ b/shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/visitor/MySQLDALVisitor.java
@@ -117,7 +117,6 @@ public final class MySQLDALVisitor extends MySQLVisitor {
         ShowColumnsStatement result = new ShowColumnsStatement();
         FromTableContext fromTableContext = ctx.fromTable();
         FromSchemaContext fromSchemaContext = ctx.fromSchema();
-        ShowLikeContext showLikeContext = ctx.showLike();
         if (null != fromTableContext) {
             FromTableSegment fromTableSegment = (FromTableSegment) visit(fromTableContext);
             result.setTable(fromTableSegment.getPattern());
@@ -126,10 +125,6 @@ public final class MySQLDALVisitor extends MySQLVisitor {
         if (null != fromSchemaContext) {
             FromSchemaSegment fromSchemaSegment = (FromSchemaSegment) visit(ctx.fromSchema());
             result.getAllSQLSegments().add(fromSchemaSegment);
-        }
-        if (null != showLikeContext) {
-            ShowLikeSegment showLikeSegment = (ShowLikeSegment) visit(ctx.showLike());
-            result.getAllSQLSegments().add(showLikeSegment);
         }
         return result;
     }

